src/components/general-table/GeneralTable.js

```diff
@@ -120,7 +120,7 @@ const GeneralTable = ({
       : apiFilterSort
       ? loadTableData(dispatch, query)
       : null;
-  }, [chipsArray, perPage, page, sortBy]);
+  }, [chipsArray, perPage, page, sortBy, hasModalSubmitted]);
```

Comment:
Adding hasModalSubmitted is causing a double render, once when it's set to true and another when it's set back to false.